test/madnlp_dense.jl

```diff
@@ -1,19 +1,79 @@
 using Test
+import MadNLP: jac_structure!, hess_structure!, obj, grad!, cons!, jac_coord!, hess_coord!, jac_dense!, hess_dense!
+using NLPModels
 using LinearAlgebra
 using SparseArrays
 using Random
 
+struct DenseQP <: AbstractNLPModel{Float64,Vector{Float64}}
```

Comment:
That structure is definitely cleaner, and by itself justifies your whole PR!!

Reply:
How about renaming it as `DenseDummyQP`? Just to convey the idea that the data are generated randomly and the problem is not very interesting, outside the scope of this tests suite?